src/components/GlobalLeaderboards/Context.tsx

```diff
@@ -0,0 +1,213 @@
+  const [accountsOrderDirection, setAccountsOrderDirection] = useState<number>(1);
+  const [positionsOrderBy, setPositionsOrderBy] = useState<keyof TopPositionsRow>("unrealizedPnl");
+  const [positionsOrderDirection, setPositionsOrderDirection] = useState<number>(1);
+  const topPositionsUnordered = useTopPositions();
+  if (
+    topPositionsUnordered &&
+    topPositionsUnordered.data &&
+    topPositionsUnordered.data.length &&
+    p.current && !p.current.positions
+  ) {
+    p.current("topPositionsUnordered");
+    p.current.positions = true;
+  }
+  const topAccountsUnordered = useTopAccounts(period);
+  if (
+    topAccountsUnordered &&
+    topAccountsUnordered.data &&
+    topAccountsUnordered.data.length &&
+    p.current && !p.current.accounts
+  ) {
+    p.current("topAccountsUnordered");
+    p.current.accounts = true;
+  }
+  const [topPositions, setTopPositions] = useState<Array<TopPositionsRow>>([]);
+  const [topAccounts, setTopAccounts] = useState<Array<TopAccountsRow>>([]);
+  const [positionsHash, setPositionsHash] = useState<string>();
+  const [accountsHash, setAccountHash] = useState<string>();
+  const topAccountsHeaderClick = useCallback((key: keyof TopAccountsRow) => () => {
+    if (key === "wins") {
+      setAccountsOrderBy(accountsOrderBy === "wins" ? "losses" : "wins");
+      setAccountsOrderDirection(1);
+    } else if (key === accountsOrderBy) {
+      setAccountsOrderDirection((d: number) => -1 * d);
+    } else {
+      setAccountsOrderBy(key);
+      setAccountsOrderDirection(1);
+    }
+  }, [accountsOrderBy]);
+
+  const topPositionsHeaderClick = useCallback((key: keyof TopPositionsRow) => () => {
+    if (key === positionsOrderBy) {
+      setPositionsOrderDirection((d: number) => -1 * d);
+    } else {
+      setPositionsOrderBy(key);
+      setPositionsOrderDirection(1);
+    }
+  }, [positionsOrderBy]);
+
+  const nextPositionsHash = utils.sha256(topPositionsUnordered.data ? topPositionsUnordered.data.map(a => (
+    a[positionsOrderBy]!.toString().split("").map((c: string) => c.charCodeAt(0))
+  )).flat() : []);
+
+  if (positionsHash !== nextPositionsHash) {
+    setPositionsHash(nextPositionsHash);
+  }
+
+  useEffect(() => {
+    if (!topPositionsUnordered.data) {
+      return;
+    }
+
+    const data = [...topPositionsUnordered.data].sort((a, b) => {
+      const key = positionsOrderBy;
+      if (a[key] instanceof BigNumber && b[key] instanceof BigNumber) {
+        return positionsOrderDirection * ((a[key] as BigNumber).gt(b[key] as BigNumber) ? -1 : 1);
+      } else {
+        // TODO: remove log
+        // eslint-disable-next-line no-console
+        console.warn("Not a BigNumber", { key, a, b });
+        return 1;
+      }
+    }).map((p, i) => ({ ...p, rank: i }));
+
+    setTopPositions(data);
+    if (p.current && !p.current.pSorted) {
+      p.current("sorted top positions");
+      p.current.pSorted = true;
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
honestly i would avoid disabling eslint in hook deps if possible, because there could be unobvious bugs after updating the code

Reply:
> honestly i would avoid disabling eslint in hook deps if possible, because there could be unobvious bugs after updating the code

@divhead I agree, but unfortunately in this case (and other similar cases) it doesn't seem possible.

`react-hooks/exhaustive-deps` expects me to provide `topPositionsUnordered.data` array as a dependency, but it makes no sense to use it here as react is not going to compare the arrays via `_.isEqual` method or similar, and will compare them by reference instead.

Because of that I cannot use the array as a dependency at all, because it always is a different object, which makes the hook run all the time and breaks the logic.

I need to re-run this hook every time any of the values we use to sort positions by changes, and to identify the change the most straightforward way is to use the values themselves.

However `react-hooks/exhaustive-deps` is not capable of figuring out the latter value is derived from the one it expects as a dependency.

Please correct me if I am wrong.

Reply:
but, the values changes when new data comes from a request? every 5 secs as other data. So, just for discussion, i thing as the whole site re-renders every 5 secs it's okay to run this hook and this is better than miss some variable in deps after changes because the rule is disabled

Reply:
I am not sure the fields of the object in the list change that often. Definitely different values change with different frequency. Say if we sort by "Win/Loss" column the values in it change only when someone closes a position and we get to know that, whereas changes in other columns are not related to the ordering. This makes me think the objects in the array may change much more often than we want to re-run this effect and reorder the rows. IMO triggering this callback only on relevant data change improves the overall performance noticeably.